### ISSUE TITLE
YAML trips over '>' in hash key [rt.cpan.org #42015]

### ISSUE THREAD

**Issue by ingydotnet** [ingydotnet/yaml-pm#94]
https://rt.cpan.org/Ticket/Display.html?id=42015

```
YAML 0.68 doesn't handle hash keys starting with a '>' character
correctly. In this example, Load() can't parse the output of a previous
Dump():

    use YAML qw(Dump Load);

    my $data = [{ ">" => 123 }];
    my $string = Dump( $data );

    Load( $string );

yields:

    YAML Error: Couldn't parse single line value
       Code: YAML_PARSE_ERR_SINGLE_LINE
       Line: 2
       Document: 1
```

**Comment by preaction** [comment]
This appears to specifically be that arrays of hashes cannot have non-word characters in the hash keys. The dumper is a lot more lenient with what can be unquotes than that part of the loader, and quoted keys are also not allowed (#36 and #48 fixed with #146).

**Comment by perlpunk** [comment]
fixed in 1.20
